oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/IterableUtils.java

```diff
@@ -0,0 +1,145 @@
+     * <p>
+     * The returned iterable has an iterator that traverses the elements in {@code a},
+     * followed by the elements in {@code b}, {@code c} and {@code d}. The source
+     * iterators are not polled until necessary.
+     * <p>
+     * The returned iterable's iterator supports {@code remove()} when the corresponding
+     * input iterator supports it.
+     *
+     * @param <E> the element type
+     * @param a  the first iterable, may not be null
+     * @param b  the second iterable, may not be null
+     * @param c  the third iterable, may not be null
+     * @param d  the fourth iterable, may not be null
+     * @return a new iterable, combining the provided iterables
+     * @throws NullPointerException if either of the provided iterables is null
+     */
+    public static <E> Iterable<E> chainedIterable(final Iterable<? extends E> a,
+                                                  final Iterable<? extends E> b,
+                                                  final Iterable<? extends E> c,
+                                                  final Iterable<? extends E> d) {
+        return org.apache.commons.collections4.IterableUtils.chainedIterable(a, b, c, d);
+    }
+
+    /**
+     * Combines the provided iterables into a single iterable.
+     * <p>
+     * The returned iterable has an iterator that traverses the elements in the order
+     * of the arguments, i.e. iterables[0], iterables[1], .... The source iterators
+     * are not polled until necessary.
+     * <p>
+     * The returned iterable's iterator supports {@code remove()} when the corresponding
+     * input iterator supports it.
+     *
+     * @param <E> the element type
+     * @param iterables  the iterables to combine, may not be null
+     * @return a new iterable, combining the provided iterables
+     * @throws NullPointerException if either of the provided iterables is null
+     */
+    @SafeVarargs
+    public static <E> Iterable<E> chainedIterable(final Iterable<? extends E>... iterables) {
+        Objects.requireNonNull(iterables);
+        return org.apache.commons.collections4.IterableUtils.chainedIterable(iterables);
+    }
+
+    /**
+     * Creates an {@code Iterable} that chains multiple {@code Iterable} instances into a single {@code Iterable}.
+     * <p>
+     * The returned {@code Iterable} will iterate over all elements of the first {@code Iterable},
+     * then all elements of the second, and so on.
+     *
+     * @param <E> the type of elements returned by the iterator
+     * @param iterables an {@code Iterable} of {@code Iterable} instances to be chained
+     * @return an {@code Iterable} that provides a single view of all elements in the input {@code Iterable} instances
+     * @throws NullPointerException if the input {@code Iterable} or any of its elements are null
+     */
+    public static <E> Iterable<E> chainedIterable(final Iterable<? extends Iterable<? extends E>> iterables) {
```

Comment:
Is this actually used anywhere in the code?

Reply:
@thomasmueller yes. 

This method would be repalcing Guava's **Iterables.concat** [1] and that method is used at many places inside code.

[1] https://guava.dev/releases/20.0/api/docs/com/google/common/collect/Iterables.html#concat-java.lang.Iterable-